### PR TITLE
WP-5635 Add CODEOWNERS file

### PR DESCRIPTION
**Pull request by evanweible-wf**
### Description

Add a `CODEOWNERS` file so that we can turn on the "require reviews from codeowners" option.
https://help.github.com/articles/enabling-required-reviews-for-pull-requests/

### Testing/QA

- [ ] CI passes

### Code Review

@Workiva/app-frameworks

### CODE REVIEW

**Comment by aviary2-wf** [comment]
## Raven
<details>
<summary>Number of Findings: 0</summary>

</details>

File: smithy.yaml [hunk 1]
```diff
@@ -1,12 +1,15 @@
 project: dart
 language: dart
 
-# dart 1.19.1
-runner_image: drydock-prod.workiva.org/workiva/smithy-runner-generator:92530
+# This image is created using https://github.com/Workiva/smithy-runner-generator
+runner_image: drydock-prod.workiva.net/workiva/smithy-runner-generator:179735 # Dart 1.24.2
```
Comment (greglittlefield-wf) [inline smithy.yaml]: If we're bumping the image, should we also bump the min SDK version to `^1.24.0`?

**Comment by evanweible-wf** [comment]
@greglittlefield-wf I raised the min SDK version to 1.24.2

**Comment by maxwellpeterson-wf** [comment]
QA +1
- [x] docs and tooling changes only
- [x] Dev +1's
- [x] CI passes
- [x] Dependency scan clean

@Workiva/release-management-pp

**Pull request merged**